src/main/java/is/citizen/sdk/crypto/CitizenCrypto.java

```diff
@@ -0,0 +1,364 @@
+        throws CryptoException {
+        Map decmap = new HashMap<>();
+        if (map != null) {
+            Iterator i = map.keySet().iterator();
+            while (i.hasNext()) {
+                String key = (String) i.next();
+                Optional<String> encval = Optional.ofNullable((String) map.get(key));
+                if (encval.isPresent()) {
+                    Optional<String> decval = decryptString(encval.get(), privKey, secret);
+                    if (decval.isPresent()) {
+                        decmap.put(key, decval.get());
+                    }
+                }
+            }
+        }
+
+        return Optional.ofNullable(decmap);
+    }
+
+    public static Optional<String> encryptString(String inputString, String publicKey)
+        throws CryptoException {
+        String encryptedText = null;
+
+        try {
+            if (StringUtils.hasText(inputString)) {
+                byte[] cipherText;
+                cipherText = encrypt(inputString.getBytes("UTF8"), getPublicKeyFromEncodedString(publicKey));
+                encryptedText = Base64.getEncoder().encodeToString(cipherText);
+            }
+        } catch (Exception e) {
+            throw new CryptoException(e.getMessage());
+        }
+
+        return Optional.ofNullable(encryptedText);
+    }
+
+
+    public static Optional<String> decryptString(String inputString, String encryptedPrivateKey, String keyPassword)
+            throws CryptoException {
+        String result = null;
+
+        try {
+            if (StringUtils.hasText(inputString)) {
+                PrivateKey privateKey = getPrivateKeyFromEncodedEncryptedString(encryptedPrivateKey, keyPassword);
+
+                byte[] decryptedText = decrypt(Base64.getDecoder().decode(inputString), privateKey);
+                result = new String(decryptedText, "UTF8");
+            }
+        } catch (IllegalBlockSizeException | IllegalArgumentException | BadPaddingException | NoSuchPaddingException
+                | NoSuchAlgorithmException | InvalidKeyException | IOException e) {
+            throw new CryptoException(e.getMessage());
+        }
+
+        return Optional.ofNullable(result);
+    }
+
+    public static boolean verifyCitizenJwt(String jwt, String encodedPublicKey) {
+        try {
+            PublicKey signingPublicKey = getPublicKeyFromEncodedString(encodedPublicKey);
+            Claims jwtClaims = Jwts.parser().setSigningKey(signingPublicKey).parseClaimsJws(jwt).getBody();
+            return jwtClaims.get(Constant.CITIZEN_JWT_AUTHENTICATED_CLAIM, Boolean.class);
+        } catch (InvalidKeySpecException | NoSuchAlgorithmException e) {
+            throw new CryptoException(e.getMessage());
+        }
+    }
+
+    /**
+     * Encrypt a text using public key.
+     *
+     * @param text The original unencrypted text
+     * @param key  The public key
+     * @return Encrypted text
+     * @throws java.lang.Exception
+     */
```

Comment:
We should add these javadoc comnments to all of the public methods at least.

Reply:
Added.